sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py

```diff
@@ -924,9 +923,9 @@ def upload_blob(self, repository: str, data: IO[bytes], **kwargs) -> Tuple[str,
                     **kwargs
                 )
             )
-        except Exception:
+        except Exception as e:
             if repository is None or data is None:
```

Comment:
Keep the `None` check here as the default error message is not clear:
- when `repository=None`, will get `ValueError: No value for given attribute` 
- when `data=None`, will get `AttributeError: 'NoneType' object has no attribute 'read'` - is unclear